### PR TITLE
Exercise for chapter 18: Inheritance

### PR DESCRIPTION
**Pull request by hothanhtuan1209**

### CODE REVIEW

File: exercise/pokerhand.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+        self.rank_hist()
+        
+        for val in self.ranks.values():
+            if val == 4:
+                return True
+        
+        return False
+         
+    def has_straight_flush(self):
+
```
Comment (cuongdoan-agilityio) [inline exercise/pokerhand.py]: Remove this empty line.

File: exercise/Card.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        """
+        Removes a card from the deck.
+
+        Parameter:
+            card (Card): The Card object to be removed from the deck.
+        """
+        
+        self.cards.remove(card)
+
+    def pop_card(self, i=-1):
```
Comment (cuongdoan-agilityio) [inline exercise/Card.py]: Update the parameter name.

File: exercise/deck_hands_card.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            hand_cards = []
+            
+            for i in range(cards_per_hand):
+                if shuffled_decks:
+                    hand_cards.append(shuffled_decks[0].pop())
+                    
+                    if not shuffled_decks[0]:
+                        shuffled_decks.pop(0)
+                else:
+                    break
```
Comment (cuongdoan-agilityio) [inline exercise/deck_hands_card.py]: Do not use variables with the same name in nested loops.

File: exercise/pokerhand.py [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+        Returns True if the hand has a flush, False otherwise.
+
+        Note that this works correctly for hands with more than 5 cards.
+        """
+        
+        self.suit_hist()
+        for val in self.suits.values():
+            if val >= 5:
+                return True
+        return False
```
Comment (cuongdoan-agilityio) [inline exercise/pokerhand.py]:
```
        self.suit_hist()

        for val in self.suits.values():
            if val >= 5:
                return True

        return False
```

File: exercise/pokerhand.py [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+        """
+        Checks if the hand has a pair.
+
+        Returns:
+            bool: True if the hand contains at least one pair (two cards of the same rank), False otherwise.
+        """
+        
+        self.rank_hist()
+        
+        for rank, count in self.ranks.items():
```
Comment (cuongdoan-agilityio) [inline exercise/pokerhand.py]: Use `_`  to naming unnecessary variables
```
for _, count in self.ranks.items():
```

File: exercise/pokerhand.py [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+            if count > 1:
+                return True
+        
+        return False
+
+    def has_twopair(self):
```
Comment (cuongdoan-agilityio) [inline exercise/pokerhand.py]: `has_two_pair`

**Comment by cuongdoan-agilityio** [comment]
For the file `exercise/pokerhand.py`
Many methods are duplicated, please clean the code.

File: exercise/pokerhand.py (hunk 1 above)
Comment (hothanhtuan1209) [reply exercise/pokerhand.py]: Done

File: exercise/Card.py (hunk 2 above)
Comment (hothanhtuan1209) [reply exercise/Card.py]: Done

File: exercise/deck_hands_card.py (hunk 3 above)
Comment (hothanhtuan1209) [reply exercise/deck_hands_card.py]: Done

File: exercise/pokerhand.py (hunk 4 above)
Comment (hothanhtuan1209) [reply exercise/pokerhand.py]: Done

File: exercise/pokerhand.py (hunk 5 above)
Comment (hothanhtuan1209) [reply exercise/pokerhand.py]: Done

File: exercise/pokerhand.py (hunk 6 above)
Comment (hothanhtuan1209) [reply exercise/pokerhand.py]: Done

**Pull request merged**